sources/packages/api/src/services/student/student.service.ts

```diff
@@ -143,27 +148,29 @@ export class StudentService extends RecordDataModelService<Student> {
         "Not able to find a student using the username (bcsc name)",
       );
     }
-
     let mustSave = false;
-
+    const userTokenBirthdate = getDateOnly(userToken.birthdate);
+    if (userToken.givenNames === undefined) {
+      userToken.givenNames = null;
+    }
```

Comment:
if(!userToken.givenNames) checks for undefined, null and empty. I'm not sure if lines 153 to 155 are necessary.

Reply:
Nice question, To all @ann-aot @andrewsignori-aot  @dheepak-aot @henryoforeh-aot @abschwenker Yes, as per the .save function in typeorm, if the value passed has undefined, the save fails for the particular column, So when the sync happens and the BCSC return the givenname as undefined, its set in our DB as null, to further creating records in the sinvalidation table. Note: The other columns are mandatory in BCSC form, so the problem is givenNames.

Reply:
Nice question, To all @ann-aot @andrewsignori-aot  @dheepak-aot @henryoforeh-aot @abschwenker Yes, as per the .save function in typeorm, if the value passed has undefined, the save fails for the particular column, So when the sync happens and the BCSC return the givenname as undefined, its set in our DB as null, to further creating records in the sinvalidation table. Note: The other columns are mandatory in BCSC form, so the problem is givenNames.

Reply:
@guru-aot, fail to save does not seems fair 😄 
It is the way that we tell Typeorm to ignore that particular column during the _update/insert_.

Reply:
He he, yeah its not doing its job properly 💯

Reply:
Not sure if we are on the same page here 😅 
Typeorm is right and it is doing what it is supposed to do.